src/Sarif/Baseline/ResultMatching/SarifLogResultMatcher.cs

```diff
@@ -273,11 +274,29 @@
                 {
                     properties = currentRun.Properties;
                 }
+
+                if (currentRun?.VersionControlProvenance != null)
+                {
+                    versionControls = versionControls.DistinctMerge(
+                        currentRun.VersionControlProvenance,
+                        VersionControlDetails.ValueComparer).ToList();
+                }
+            }
+
+            foreach (Run previousRun in previousRuns ?? Enumerable.Empty<Run>())
+            {
+                if (previousRun?.VersionControlProvenance != null)
+                {
+                    versionControls = versionControls.DistinctMerge(
+                        previousRun.VersionControlProvenance,
+                        VersionControlDetails.ValueComparer).ToList();
+                }
             }
 
             run.Graphs = graphs;
             run.Invocations = invocations;
             run.Properties = properties;
+            run.VersionControlProvenance = versionControls.Any() ? versionControls : null;
```

Comment:
>versionControls [](http://example.com/codeflow?start=43&length=15)

Just use versionControls.FirstOrDefault() here.

Reply:
This check is to prevent `run.VersionControlProvenance` to be set to an empty list. 

If it's an empty list, the output sarif file contains the entry.
```
      "versionControlProvenance": [ ],
```
if it's null the output file doesn't contain it.

The empty and null VersionControlProvenance are same semantically, but the empty entry will break some file diffing tests. What do you think?